fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java

```diff
@@ -199,15 +198,20 @@ public Scope visitSelect(SelectRelation selectRelation, Scope scope) {
             sourceScope.setParent(scope);
 
             Map<Expr, SlotRef> generatedExprToColumnRef = new HashMap<>();
-            if (!(resolvedRelation instanceof ViewRelation)) {
-                new AstTraverser<Void, Void>() {
-                    @Override
-                    public Void visitTable(TableRelation tableRelation, Void context) {
-                        generatedExprToColumnRef.putAll(tableRelation.getGeneratedExprToColumnRef());
-                        return null;
-                    }
-                }.visit(resolvedRelation);
-            }
+            new AstVisitor<Void, Void>() {
+                @Override
+                public Void visitTable(TableRelation tableRelation, Void context) {
+                    generatedExprToColumnRef.putAll(tableRelation.getGeneratedExprToColumnRef());
+                    return null;
```

Comment:
Do we need process other relation type?

Reply:
No, currently only direct rewriting of generated columns is supported.